client/test/e2e/arc/navigation.spec.js

```diff
@@ -0,0 +1,131 @@
+  it('should login,' +
+    ' navigation to API Composer view,' +
+    ' and log out',
+    function() {
+
+      var loginView = new ArcViews.LoginView();
+      var landingView = new ArcViews.LandingView();
+      var composerHomeView = new ComposerViews.ComposerHomeView();
+      var headerView = new ArcViews.HeaderView();
+
+      loginView.loginToLandingView();
+      landingView.openComposerView();
+      expect(composerHomeView.componentIdentifier.getText())
+        .toEqual('Composer');
+
+      headerView.logout();
+    });
+});
+
+describe('build-deploy-navigation', function() {
+  it('should login,' +
+    ' navigation to Build Deploy view,' +
+    ' and log out',
+    function() {
+
+      var loginView = new ArcViews.LoginView();
+      var landingView = new ArcViews.LandingView();
+      var buildDeployHomeView = new BuildDeployViews.BuildDeployHomeView();
+      var headerView = new ArcViews.HeaderView();
+
+      loginView.loginToLandingView();
+
+      landingView.openBuildDeployView();
+      expect(buildDeployHomeView.componentIdentifier.getText())
+        .toEqual('Build & Deploy');
+      headerView.logout();
+    });
+});
+
+
+describe('process-manager-navigation', function() {
+  it('should login,' +
+    ' navigation to Process Manager view,' +
+    ' and log out',
+    function() {
+
+      var loginView = new ArcViews.LoginView();
+      var landingView = new ArcViews.LandingView();
+      var processManagerHomeView = 
+        new ProcessManagerViews.ProcessManagerHomeView();
+      var headerView = new ArcViews.HeaderView();
+
+      loginView.loginToLandingView();
+
+      landingView.openProcessManagerView();
+      expect(processManagerHomeView.componentIdentifier.getText())
+        .toEqual('Process Manager (Early Access)');
+      headerView.logout();
+    });
+});
+
+describe('metrics-navigation', function() {
+  it('should login,' +
+    ' navigation to Build Deploy view,' +
+    ' and log out',
+    function() {
+
+      var loginView = new ArcViews.LoginView();
+      var landingView = new ArcViews.LandingView();
+      var metricsHomeView = new MetricsViews.MetricsHomeView();
+      var headerView = new ArcViews.HeaderView();
+
+      loginView.loginToLandingView();
+
+      landingView.openMetricsView();
+      expect(metricsHomeView.componentIdentifier.getText())
+        .toEqual('Metrics');
+      headerView.logout();
+    });
+});
+
+
+describe('tracing-navigation', function() {
+  it('should login,' +
+    ' navigation to Tracing view,' +
+    ' and log out',
+    function() {
+
+      var loginView = new ArcViews.LoginView();
+      var landingView = new ArcViews.LandingView();
+      var tracingHomeView = new TracingViews.TracingHomeView();
+      var headerView = new ArcViews.HeaderView();
+
+      loginView.loginToLandingView();
+
+      landingView.openTracingView();
+      expect(tracingHomeView.componentIdentifier.getText())
+        .toEqual('Tracing (Early Access)');
```

Comment:
this may be a bit fragile as the text is likely to change over time.

Relying on any text in the ui should be seen as potentialy fragile.

We really should be using tokenized strings and verifying the token as a HTML attribute of the target element